enterprise/frontend/src/embedding-sdk/store/auth.ts

```diff
@@ -0,0 +1,159 @@
+import type {
+  EmbeddingSessionToken,
+  FetchRequestTokenFn,
+  SDKConfig,
+} from "embedding-sdk";
+import type { SdkStoreState } from "embedding-sdk/store/types";
+import api from "metabase/lib/api";
+import { createAsyncThunk } from "metabase/lib/redux";
+import { refreshSiteSettings } from "metabase/redux/settings";
+import { refreshCurrentUser } from "metabase/redux/user";
+
+import { getOrRefreshSession } from "./reducer";
+import { getFetchRefreshTokenFn } from "./selectors";
+
+export const initAuth = createAsyncThunk(
+  "sdk/token/INIT_AUTH",
+  async (sdkConfig: SDKConfig, { dispatch }) => {
+    // Setup JWT or API key
+    const isValidJwtConfig =
+      sdkConfig.jwtProviderUri && sdkConfig.jwtProviderUri?.length > 0;
+    const isValidApiKeyConfig =
+      sdkConfig.apiKey && window.location.hostname === "localhost";
+
+    if (isValidJwtConfig) {
+      // JWT setup
+      api.onBeforeRequest = async () => {
+        const session = await dispatch(
+          getOrRefreshSession(sdkConfig.jwtProviderUri!),
+        ).unwrap();
+        if (session?.id) {
+          api.sessionToken = session.id;
+        }
+      };
+      // verify that the jwt is actually valid before proceeding
```

Comment:
This is so that if the session is not valid, it's the `initAuth` thunk that fails, and not the calls to `/user/current` or to `/properties`. This should make for a slightly better output in the terminal